spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/batch/BasicBatchConfigurer.java

```diff
@@ -42,13 +42,13 @@
  */
 public class BasicBatchConfigurer implements BatchConfigurer, InitializingBean {
 
-	private final BatchProperties properties;
+	protected final BatchProperties properties;
```

Comment:
Generally we prefer to keep member variables `private`. Is there a reason you changed these?

Reply:
From the design of this class it looks like it it's open for extension (see `protected` methods). However, subclasses cannot access the members those protected methods operate on, even though it's passed to their constructor. I'd suggest either extend the visibility to `protected` or provide protected getters to these. Since the members are `final` I think it's safe and would improve extensibility. What do you think?

For this concrete member, the new `JpaBatchConfigurer` implementation uses the `properties` too in `determineIsolationLevel`

Reply:
Given that the fields don't need to be made `protected` to implement the configurable isolation level, I think they should be considered via a separate issue please. Even `properties` could remain `private` as `JpaBatchConfigurer` could capture what it needs in its constructor. This would minimise the changes to the public API.

Reply:
Sorry for the delay. Sure, I'll roll this change back and will consider a separate PR for this.